fabric8-maven-plugin/src/main/java/io/fabric8/maven/ApplyMojo.java

```diff
@@ -191,6 +191,7 @@ public void execute() throws MojoExecutionException, MojoFailureException {
             // lets check we have created the namespace
             String namespace = getNamespace();
             controller.applyNamespace(namespace);
+            controller.setNamespace(namespace);
```

Comment:
applyNamespace() creates a namespace object if it doesn't already exists.
setNamespace() its just a setter.

The controller carries state and in some cases it becomes stale. That's why we also need to setNamespace().